### PR TITLE
Rewrite async actions to use Nan::AsyncWorker

### PR DESCRIPTION
**Pull request by klokane**
Missing validate, lost while rebase. Will follow in next commit

### CODE REVIEW

**Comment by klokane** [comment]
ready for review

File: src/validate_async.cc [hunk 1]
```diff
@@ -6,69 +6,130 @@ using std::string;
...
-            drafter_free_result(result);
-        }
-    }
-    // Callback
-    Nan::Persistent<Function> callback;
-
-    // Input
+class ValidateWorker : public AsyncWorker {
     drafter_parse_options parseOptions;
+    drafter_serialize_options serializeOptions;
```
Comment (pksunkara) [inline src/validate_async.cc]: You don't need this here.

File: src/validate_async.cc [hunk 2]
```diff
@@ -6,69 +6,130 @@ using std::string;
...
+        }
+
+        // v8::Local<v8::Value> argv[] = {
+        //    Nan::Null(),
+        //    refract2v8(result, serializeOptions)
+        //};
+
+        // if (0 != parse_err_code) {
+        //    argv[0] = annotations2v8(result);
+        //}
```
Comment (pksunkara) [inline src/validate_async.cc]: Let's remove this.

Comment (klokane) [reply src/validate_async.cc]: Sry, I remove it, but it seem I forgot to push it

**Comment by klokane** [comment]
@pksunkara is it ready for merge?

**Comment by klokane** [comment]
Updated

**Pull request merged**